iothub/device/src/Message.cs

```diff
@@ -326,6 +326,32 @@ public byte[] GetBytes()
             return ReadFullStream(_bodyStream);
         }
 
+        /// <summary>
+        /// Clones an existing <see cref="Message"/> instance and sets content body defined by <paramref name="byteArray"/> on it.
+        /// </summary>
+        /// <param name="byteArray">Message content to be set after clone.</param>
+        /// <returns>A new instance of <see cref="Message"/> with body content defined by <paramref name="byteArray"/>,
+        /// and user/system properties of the cloned <see cref="Message"/> instance.
+        /// </returns>
+        /// <remarks>user should treat the input byte array as immutable when
+        /// sending the message.</remarks>
+        public Message CloneWithBody(in byte[] byteArray)
+        {
+            var result = new Message(byteArray);
+
+            foreach (string key in this.Properties.Keys)
```

Comment:
we've chosen to not use "this." where not necessary. The rest of this file looks to be following that mode, so you can remove them here too.

Reply:
removed 'this.'